src/log.c

```diff
@@ -267,11 +267,14 @@ static void refsIncr(struct raft_log *l,
 
     /* Lookup the slot associated with the given term/index, which must have
      * been previously inserted. */
-    for (slot = &l->refs[key]; slot != NULL; slot = slot->next) {
+    slot = &l->refs[key];
+    while (1) {
```

Comment:
This is safe to do because, by construction, we know that we'll find a slot with `slot->term == term` and the loop will terminate. This new version avoids comparing to `slot` to `NULL` which makes Coverity unhappy (since the code later dereferences `slot`).